crates/symbolicator-proguard/src/java.rs

```diff
@@ -0,0 +1,229 @@
+        return ty.clone();
+    } else if token == 'L' {
+        // invalid signature
+        let l = chrs.clone().last();
+        if l.is_none() || l.unwrap() != ';' {
+            return "".to_string();
+        }
+        chrs.next_back(); // remove final `;`
+        let obfuscated = chrs.collect::<String>().replace('/', ".");
+        // if mappers.len() > 0 {}
+        for mapper in mappers {
+            if let Some(mapped) = mapper.remap_class(&obfuscated) {
+                return mapped.to_string();
+            }
+        }
+        return obfuscated;
+    } else if token == '[' {
+        return format!(
+            "{}[]",
+            byte_code_type_to_java_type(chrs.collect::<String>(), mappers)
+        );
+    }
+    byte_code_type
+}
+
+// parse_obfuscated_bytecode_signature will parse an obfuscated signatures into parameter
+// and return types that can be then deobfuscated
+fn parse_obfuscated_bytecode_signature(signature: &str) -> Option<(Vec<String>, String)> {
+    let mut chrs = signature.chars();
+
+    let token = chrs.next();
+    if token.unwrap_or_default() != '(' {
+        return None;
+    }
+
+    let sig = chrs.collect::<String>();
+    let split_sign = sig.rsplitn(2, ')').collect::<Vec<&str>>();
+    if split_sign.len() != 2 {
+        return None;
+    }
+
+    let return_type = split_sign[0];
+    let parameter_types = split_sign[1];
+    if return_type.is_empty() {
+        return None;
+    }
+
+    let mut types: Vec<String> = Vec::new();
+    let mut tmp_buf: Vec<char> = Vec::new();
+
+    let mut param_chrs = parameter_types.chars();
+    while let Some(token) = param_chrs.next() {
+        if java_base_types().contains_key(&token.to_string()) {
+            if !tmp_buf.is_empty() {
+                tmp_buf.push(token);
+                types.push(tmp_buf.iter().collect());
+                tmp_buf.clear();
+            } else {
+                types.push(token.to_string());
+            }
+        } else if token == 'L' {
+            tmp_buf.push(token);
+            for c in param_chrs.by_ref() {
+                tmp_buf.push(c);
+                if c == ';' {
+                    break;
+                }
+            }
+            if tmp_buf.is_empty() || !tmp_buf.ends_with(&[';']) {
+                return None;
+            }
+            types.push(tmp_buf.iter().collect());
+            tmp_buf.clear();
+        } else if token == '[' {
+            tmp_buf.push('[');
+        } else {
+            tmp_buf.clear();
+        }
+    }
+
+    Some((types, return_type.to_string()))
+}
+
+// returns a tuple where the first element is the list of the function
+// parameters and the second one is the return type
+pub fn deobfuscate_bytecode_signature(
+    signature: &str,
+    mappers: &[&proguard::ProguardMapper],
+) -> Option<(Vec<String>, String)> {
+    if let Some((parameter_types, return_type)) = parse_obfuscated_bytecode_signature(signature) {
```

Comment:
```suggestion
    let (parameter_types, return_type) = parse_obfuscated_bytecode_signature(signature)?;
```

You can use the `?` operator to propagate `None`.